torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -120,44 +123,161 @@ def can_auto_functionalize(op: OperatorBase) -> bool:
     return True
 
 
+@dataclass
+class ViewInfo:
+    base_index: int
+    size: Any = None
+    stride: Any = None
+    storage_offset: Any = None
+    is_alias_to_base: bool = False
```

Comment:
These would do well with some comments. For example, when is_alias_to_base is False, then size/stride/storage_offset are None?

Reply:
addressed.